src/ReportManager/DetailsSection/SchemaForm/index.js

```diff
@@ -15,14 +17,51 @@ export const FIELDS = {
   [FORM_ELEMENT_TYPES.TEXT]: Text,
 };
 
-const SchemaForm = ({ formData, onFormChange, onFormSubmit, renderSubmitButton, schema }) => {
-  const fields = useMemo(() => schema ? makeFieldsFromSchema(schema) : {}, [schema]);
+const ajv = new Ajv2020({ allErrors: true });
+
+const SchemaForm = ({ formData, onFormDataChange, onFormSubmit, renderSubmitButton, schema }) => {
+  const { t } = useTranslation('reports', { keyPrefix: 'reportManager.detailsSection.schemaForm' });
+
+  const [fieldErrors, setFieldErrors] = useState({});
+
+  const fields = useMemo(() => makeFieldsFromSchema(schema), [schema]);
+
+  const fieldValues = useMemo(() => Object.entries(formData).reduce((accumulator, [fieldId, fieldValue]) => {
```

Comment:
Here we memoize the field values in a flat object. Every time the form data is updated this object will too.